src/tools/nbench/nbench.cpp

```diff
@@ -33,6 +35,121 @@
 using namespace std;
 using namespace ngraph;
 
+class PerfShape : public ngraph::runtime::PerformanceCounter
+{
+public:
+    PerfShape(const runtime::PerformanceCounter& p, Shape s)
+        : PerformanceCounter(p)
+        , shape(s)
+    {
+    }
+    Shape shape;
+};
+
+unordered_map<string, shared_ptr<Node>> get_node_map(shared_ptr<Function> func)
+{
+    unordered_map<string, shared_ptr<Node>> node_map;
+    vector<shared_ptr<Function>> fs;
+    traverse_functions(func, [&](shared_ptr<Function> f) { fs.push_back(f); });
+    for (shared_ptr<Function> f : fs)
+    {
+        for (shared_ptr<Node> node : f->get_ops())
+        {
+            node_map.insert({node->get_name(), node});
+        }
+    }
+    return node_map;
+}
+
+vector<PerfShape> to_perf_shape(shared_ptr<Function> f,
+                                const vector<runtime::PerformanceCounter>& perf_data)
+{
+    vector<PerfShape> result;
+    auto node_map = get_node_map(f);
+    for (const runtime::PerformanceCounter& p : perf_data)
+    {
+        auto node = node_map[p.name()];
+        Shape shape = node->get_outputs()[0].get_shape();
+        result.push_back(PerfShape(p, shape));
+    }
+    return result;
+}
+
+multimap<size_t, string> aggregate_timing_details(const vector<PerfShape>& perf_data)
+{
+    unordered_map<string, size_t> timing;
+    unordered_map<string, size_t> count;
+    for (const PerfShape& p : perf_data)
+    {
+        string op = p.name().substr(0, p.name().find('_'));
+        string shape_name = " {" + join(p.shape) + "} ";
+        timing[op + shape_name] += p.microseconds();
```

Comment:
Initialization?

Reply:
unordered_map will create an entry if one does not exist and initialize it to zero or the default ctor. `timinig["some string not in map"] == 0`

Reply:
COOL. Did know that.